_drafts/intro-to-working-with-data-in-CUBA.adoc

```diff
@@ -0,0 +1,201 @@
+
+
+== What will be build
+
+This guide enhances the https://github.com/cuba-platform/cuba-petclinic[CUBA petclinic] example to show different use cases regarding programmatic data access with the DataManager API. In particular, the following use cases will be covered:
+
+* automatic creation of a Visit for a given Pet Identification Number
+* automatic sending out of "Disease Warning Mailing" for endangered pets
+
+=== Final Application
+image:https://dummyimage.com/600x400/00b5ec/ffffff.png&text=CUBA Petclinic Overview[] Overview of final CUBA petclinic application
+
+
+include::includes/guide_requirements.adoc[]
+
+
+
+== Programmatic data access interactions
+
+Oftentimes it is necessary to interact with the database in a programmatic manner. In a CUBA based application, these cases are not as omnipresent as in other frameworks due to its declarative nature of data-binding for UIs, but still, there are
+a lot of cases where data has to be loaded from the database or stored in the database as part of a business process.
+
+The main interaction point that CUBA provides is the https://doc.cuba-platform.com/manual-6.10/dataManager.html[DataManager] API. This API allows the developer of a CUBA application to load and store data in the database through the general abstraction of entities.
+
+DataManager is an Abstraction on top of the Java Persistence API (JPA). It hides away certain details that are not required on a day-to-day basis.
+
+It also adds certain capabilities that are not available out of the box in JPA. Those capabilities are mainly the https://doc.cuba-platform.com/manual-6.10/views.html[Views concept] as well as the application of https://doc.cuba-platform.com/manual-6.10/security_subsystem.html[security constrains]. More information about differences between JPA and DataManager can be found in the docs on https://doc.cuba-platform.com/manual-6.10/dm_vs_em.html[DataManager vs. EntityManager].
+
+
+include::includes/petclinic_introduction.adoc[]
+
+
+== Load data with DataManager
+
+DataManager has multiple ways to load data from the database. The following examples contain the most common cases for loading data in different ways.
+
+=== Single Item
+
+The first option is to load a single Entity instance from the database. This option is often required if reference data needs to be read from the database in order to execute some logic for that entity instance or with the help of that entity instance.
+
+In the case of the Petclinic example, there is some business logic which should create a new Visit for a given Pet by its ID.
+
+image:https://dummyimage.com/600x400/00b5ec/ffffff.png&text=Create visit for Pet[] Create visit for Pet overview
+
+
+The part that loads the Pet instance by its ID looks like this with the use of the DataManager API:
+
+
+.VisitServiceBean.java
+[source,java]
+----
+@Inject
+DataManager dataManager;
+
+Pet loadPetById(String petId) {
```

Comment:
Is there any reason to make this method package-private?

Reply:
no - it is actually an old snippet. Will update with "IdentificationNumber" example